### PR TITLE
Recipe `IllegalArgumentExceptionToAlreadyConnectedException`

### PR DESCRIPTION
**Pull request by BhavanaPidapa**
## What's changed?
This PR contains recipe - org.openrewrite.java.migrate.MethodExceptionReplacerRecipe
Rules:
image

image

## What's your motivation?
A generic custom recipe for exception replacement based on method signatures. A common recipe is written to handle to both the rules.

## Anything in particular you'd like reviewers to focus on?
image
For this rule already a custom recipe is written but now making use of this generic recipe instead of the previous one.

## Anyone you would like to review specifically?
@timtebeek @cjobinabo 

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/migrate/MethodExceptionReplacerRecipe.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Map;
+
+public class MethodExceptionReplacerRecipe extends Recipe {
+    private final Map<String, Map<String, String>> methodToExceptionMapping;
+
+    public MethodExceptionReplacerRecipe(Map<String, Map<String, String>> methodToExceptionMapping) {
+        this.methodToExceptionMapping = methodToExceptionMapping;
+    }
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/MethodExceptionReplacerRecipe.java:35]: We don't support Map options; you can see this fail in the validation; can this be revised to individual `@Option` fields? Or a dedicated recipe as opposed to a generic one?
```
        Missing{property='MethodExceptionReplacerRecipe.methodToExceptionMapping', value='null', message='is required'}],
        Missing{property='MethodExceptionReplacerRecipe.methodToExceptionMapping', value='null', message='is required'}]
```

Comment (BhavanaPidapa) [reply src/main/java/org/openrewrite/java/migrate/MethodExceptionReplacerRecipe.java]: Created a dedicated Recipe.

**Review by timtebeek — APPROVED** [review]
Thanks so far @BhavanaPidapa ! I've pushed up a small change for a missing import after the migration; figured that'd help ensure the code compiles after the recipe runs.

In the PR description you also mention Class.getAnnotation, but I do not see that mentioned in the recipe or tests; is that still to come?

**Comment by BhavanaPidapa** [comment]
It is already covered in https://github.com/openrewrite/rewrite-migrate-java/pull/613

**Comment by timtebeek** [comment]
Ok; good to merge then. I know @cjobinabo is out; let me know if you have any concerns or difficulties merging.

**Comment by BhavanaPidapa** [comment]
> Ok; good to merge then. I know @cjobinabo is out; let me know if you have any concerns or difficulties merging.

Hey Tim, I don't have access to merge it seems. Can you please provide me the access.

**Pull request merged**